ttnn/cpp/ttnn/operations/pool/rotate/device/kernels/dataflow/reader_rotate_nearest_interleaved.cpp

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+#include <cstdint>
+#include <dataflow_api.h>
+#include <ttnn/cpp/ttnn/operations/pool/device/kernels/pool_kernels_common.hpp>
+#include <ttnn/cpp/ttnn/operations/pool/rotate/device/kernels/fixed_point_q16.h>
+
+// Maximum number of sticks to process per batch for optimal NOC utilization
+constexpr uint32_t MAX_BATCH_SIZE = 5;
+
+void kernel_main() {
+    // Runtime arguments
+    uint32_t input_addr = get_arg_val<uint32_t>(0);
+    uint32_t num_sticks = get_arg_val<uint32_t>(1);
+    uint32_t start_stick_id = get_arg_val<uint32_t>(2);
+    fixed_point_t cos_angle = static_cast<fixed_point_t>(get_arg_val<uint32_t>(3));
+    fixed_point_t sin_angle = static_cast<fixed_point_t>(get_arg_val<uint32_t>(4));
+    fixed_point_t center_x = static_cast<fixed_point_t>(get_arg_val<uint32_t>(5));
+    fixed_point_t center_y = static_cast<fixed_point_t>(get_arg_val<uint32_t>(6));
+    uint32_t fill_value_bf16 = get_arg_val<uint32_t>(7);
+
+    // Compile-time arguments
+    constexpr uint32_t output_cb_index = get_compile_time_arg_val(0);
+    constexpr uint32_t input_stick_nbytes = get_compile_time_arg_val(1);
+    constexpr uint32_t input_batch = get_compile_time_arg_val(2);
+    constexpr uint32_t input_height = get_compile_time_arg_val(3);
+    constexpr uint32_t input_width = get_compile_time_arg_val(4);
+    constexpr uint32_t input_channels = get_compile_time_arg_val(5);
+    constexpr uint32_t num_cb_pages = get_compile_time_arg_val(6);
+    constexpr uint32_t fill_cb_index = get_compile_time_arg_val(7);
+    constexpr uint32_t input_stick_nbytes_unaligned = get_compile_time_arg_val(8);
+    constexpr bool fill_is_zero = get_compile_time_arg_val(9) != 0;
+
+    constexpr auto src_args = TensorAccessorArgs<10>();
+    const auto input_tensor_accessor = TensorAccessor(src_args, input_addr, input_stick_nbytes);
+    // Batch size is limited by available CB pages or MAX_BATCH_SIZE, whichever is smaller
+    constexpr uint32_t batch_size = num_cb_pages < MAX_BATCH_SIZE ? num_cb_pages : MAX_BATCH_SIZE;
```

Comment:
let's move this batch size to factory and pass in final batch to both reader/writer to not risk they go out of sync or CB allocations pattern in factory changes this